core/provider/service.go

```diff
@@ -450,14 +453,42 @@ func (s *Service) getResources(ctx context.Context, p *domain.Provider) ([]*doma
 		return nil, err
 	}
 
+	resourceTypeFilterMap := make(map[string]string)
+	for _, rc := range p.Config.Resources {
+		if len(rc.Filter) > 0 {
+			resourceTypeFilterMap[rc.Type] = rc.Filter
+		}
+	}
 	res, err := c.GetResources(p.Config)
 	if err != nil {
 		return nil, fmt.Errorf("error fetching resources for %v: %w", p.ID, err)
 	}
 
+	filteredResources := make([]*domain.Resource, 0)
+	for _, r := range res {
+		if filterExpression, ok := resourceTypeFilterMap[r.Type]; ok {
+			resourceMap, err := structToMap(r)
+			if err != nil {
+				return nil, fmt.Errorf("parsing resource struct to map: %w", err)
+			}
+
+			v, err := evaluator.Expression(filterExpression).EvaluateWithVars(map[string]interface{}{
+				"resource": resourceMap,
+			})
+			if err != nil {
+				return nil, err
+			}
+			if !reflect.ValueOf(v).IsZero() {
```

Comment:
I'm not sure if this also checks if `v` is nil. Can you please check and add a `IsNil` check for `v`?

Reply:
ValueOf check for nil